optashift-employee-rostering-webapp/src/test/java/org/optaplanner/openshift/employeerostering/webapp/skill/AbstractRestServiceIT.java

```diff
@@ -0,0 +1,56 @@
+package org.optaplanner.openshift.employeerostering.webapp.skill;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response.Status;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.junit.Before;
+import org.optaplanner.openshift.employeerostering.client.ServiceClientFactory;
+import org.optaplanner.openshift.employeerostering.webapp.tools.ClientResponseContextAssert;
+import org.optaplanner.openshift.employeerostering.webapp.tools.RecordingClientResponseFilter;
+import org.optaplanner.openshift.employeerostering.webapp.tools.TestConfig;
+
+public class AbstractRestServiceIT {
+
+    private static final String BASE_TEST_URL = TestConfig.getApplicationUrl();
+
+    //Consequence of reusing JPA entities on client - cannot rely on equals()
+    protected static final  String [] IGNORED_FIELDS = {"id", "version"};
+
+    protected final ServiceClientFactory serviceClientFactory;
+    protected final RecordingClientResponseFilter recordingClientResponseFilter;
+
+    protected AbstractRestServiceIT() {
+        URL baseTestUrl;
+        try {
+            baseTestUrl = new URL(BASE_TEST_URL);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("Base URL is invalid.", e);
```

Comment:
throw new IllegalArgumentException("The baseTestUrl (" + baseTestUrl + ") is invalid.", e);